libyul/AsmParser.cpp

```diff
@@ -109,7 +119,7 @@ Statement Parser::parseStatement()
 			fatalParserError(4904_error, "Case not allowed after default case.");
 		if (_switch.cases.empty())
 			fatalParserError(2418_error, "Switch statement without any cases.");
-		_switch.location.end = _switch.cases.back().body.location.end;
+		_switch.debugData = updateLocationEndFrom(_switch.debugData, _switch.cases.back().body.debugData->location);
```

Comment:
Why not use the helpers here?
```suggestion
		_switch.debugData = updateLocationEndFrom(debugDataOf(_switch), locationOf(_switch.cases.back().body));
```

There are also a few similar places below.

Reply:
In places where I wasn't able to use it, fails with:
`error: no member named 'valueless_by_exception' in 'solidity::yul::Switch'`